src/PrettyConvert/converters.js

```diff
@@ -91,27 +104,38 @@ export const converters: { [string]: ?Function } = {
     }
     return prettyConvert(resolveFromGeneric(type), components);
   },
-  object: (type: K.Obj, components: Components, depth: number) => (
-    <span>
-      <components.TypeMeta>
-        Shape <components.Outline>{'{'}</components.Outline>
-      </components.TypeMeta>
-      <components.Indent>
-        {type.members.map(prop => {
-          if (prop.kind === 'spread') {
-            const nestedObj = resolveFromGeneric(prop.value);
-            return nestedObj.members.map(newProp =>
-              prettyConvert(newProp, components, depth),
-            );
-          }
-          return prettyConvert(prop, components, depth);
-        })}
-      </components.Indent>
-      <components.TypeMeta>
-        <components.Outline>{'}'}</components.Outline>
-      </components.TypeMeta>
-    </span>
-  ),
+  object: (type: K.Obj, components: Components, depth: number) => {
+    if (type.members.length === 0) {
+      return <components.Type>object</components.Type>;
+    }
+    let simpleObj =
+      type.members.filter(mem => !SIMPLE_TYPES.includes(mem.kind)).length === 0;
+
+    if (simpleObj) {
+      return <components.Type>{convert(type)}</components.Type>;
+    }
+    return (
+      <span>
```

Comment:
Only complex objects use the complex 'shape' renderer.

I'm close to pulling out the word 'shape' here as it never seems useful in actual docs. It's not really informative unless you already understand it.